### ISSUE TITLE
Printing loss after first training step - Keras with Tensorflow backend

### ISSUE THREAD

**Issue by SrivastavaKshitij** [keras-team/keras#7933]
Hi there

I am training my Deep neural network on Keras (TF backend). I just want to print the first loss while training DNN. I just want to make sure that my initialization is correct and so I need the initial loss calculated by the DNN after making the first forward pass.

Keras callback allows us to determine loss after every epoch. I want it after first training step.
Any help is deeply appreciated. 

Thanks

**Comment by yun-long** [comment]
Maybe you can try model.fit_generator or model.train_on_batch. And make the batch size = 1, then the loss after epoch 1 is the one you want.
